single/sol/sol.hpp

```diff
@@ -6899,16 +6906,24 @@ namespace sol {
 		struct check_getter<T, std::enable_if_t<std::is_integral<T>::value && lua_type_of<T>::value == type::number>> {
 			template <typename Handler>
 			static optional<T> get(lua_State* L, int index, Handler&& handler, record& tracking) {
-				int isnum = 0;
-				lua_Integer value = lua_tointegerx(L, index, &isnum);
-				if (isnum == 0) {
-					type t = type_of(L, index);
-					tracking.use(static_cast<int>(t != type::none));
-					handler(L, index, type::number, t);
-					return nullopt;
+#if SOL_LUA_VERSION >= 503
+				if (lua_isinteger(L, index) == 1) {
+					tracking.use(1);
+					return static_cast<T>(lua_tointeger(L, index));
+				} 
```

Comment:
lua_isinteger followed by lua_tointeger is a slight inefficiency: lua_tointegerx already checks if the number is an integer and performs the cast and `lua_tointeger` re-does this work in 5.3 since it's just a macro around `lua_tointegerx`.